.ci/generate_bear_requirements.py

```diff
@@ -144,6 +154,49 @@ def get_pip_requirements(requirements):
     return _get_requirements(requirements, '~=', inherited_requirements)
 
 
+def deep_update(target, src):
+    for key, value in src.items():
+        if not key in target:
+            target[key] = copy.deepcopy(value)
+        else:
+            if isinstance(value, list):
+                target[key].extend(value)
+            elif isinstance(value, dict):
+                deep_update(target[key], value)
+            elif isinstance(value, set):
+                target[key].update(value.copy())
+            else:
+                target[key] = copy.copy(value)
+
+
+def deep_diff(target, src):
+    errors = []
+    for key, value in src.items():
+        if not key in target:
+                errors.append((key, 'Missing'))
+        elif target[key] != value:
+            if isinstance(value, list):
+                if [x for x in value if x not in target[key]]:
+                    errors.append(key)
+            elif isinstance(value, dict):
+                if target[key] != value:
+                    e = deep_diff(target[key], value)
+                    errors.append((key, e))
+            elif isinstance(value, set):
+                if set(target[key]).symmetric_difference(value):
+                    errors.append(key)
+            else:
+                errors.append((key, target[key]))
+    return errors
+
+
+def sort_requirements(req_dict):
```

Comment:
@jayvdb This was something that I missed earlier. Now the generated file will always have reqs in sorted order.

Reply:
cant this be done in the iterator of `_get_requirements` ?

Reply:
No, the keys added by `deep_update`, are not sorted as new keys are appended to the end. This is sorts them at the end of all processing.